### PR TITLE
Add TSLint rule file and fix all violations

### PR DESCRIPTION
**Pull request by BrennanConroy**
Biggest things to look for are accidentally making functions public that shouldn't be and of course behavior changes (which there shouldn't be)

### CODE REVIEW

**Comment by anurse** [comment]
Also, run the functional tests

**Comment by BrennanConroy** [comment]
Don't worry @anurse, I ran the Functional tests

**Comment by davidfowl** [comment]
Why is appveyor sad?

**Comment by anurse** [comment]
> Why is appveyor sad?

I'll give you a hint, it rhymes with "shlakey best"

File: client-ts/package.json [hunk 1]
```diff
@@ -18,7 +18,8 @@
     "rollup-plugin-commonjs": "^8.2.6",
     "rollup-plugin-node-resolve": "^3.0.2",
     "rollup-plugin-sourcemaps": "^0.4.2",
-    "typescript": "^2.6.2",
+    "tslint": "^5.9.1",
```
Comment (BrennanConroy) [inline client-ts/package.json]: @Eilon We're adding a new dev dependency
@anurse will submit an OSS report for it once approved

Comment (Eilon) [reply client-ts/package.json]: Sounds good!

**Comment by anurse** [comment]
Rebase?

**Comment by BrennanConroy** [comment]
🆙 📅

**Comment by anurse** [comment]
I won the race!

screen shot 2018-02-07 at 9 42 46 am

**Comment by BrennanConroy** [comment]
🆙 :date: Rebase
Functional tests passed

**Pull request merged**